Refactor: Movement history structure and controller fixes for Cost, Payment, GeneralExpense, and Debt

SJAmanda opened the pull request:
Description

This pull request introduces structural improvements to the movement history system and applies necessary fixes to controllers, models, and report views to ensure data consistency and proper locality handling.

Changes Overview

🧩 Controllers

CostController.php – Ensured correct filtering by locality_id and consistent created_by relationships.

PaymentController.php – Applied whereHas filtering for locality validation.

GeneralExpenseController.php – Adjusted query logic for locality consistency.

DebtController.php – Implemented the same filtering logic for accuracy.

MovementHistoryController.php – Refactored to support new grouped report structure.

🎨 Views

resources/views/localities/index.blade.php – Updated the checkbox to use AdminLTE style for better UI consistency.

resources/views/reports/pdfMovementsHistory.blade.php – Adjusted layout and grouping logic for reports.

🆕 New Views Added

pdfMovementsHistoryGroupedByModule.blade.php – Displays the movement history grouped by module.

pdfMovementsHistoryModuleTitle.blade.php – Partial view for grouped module headers.

🔗 Routes

routes/web.php – Added and adjusted routes to support new report structure and controller logic.

SJAmanda committed:
Refactor: structure changes in movement history and fixes in Cost, Payment, GeneralExpense, and Debt controllers

And-Y21 commented on app/Http/Controllers/GeneralExpenseController.php:
```diff
@@ -15,10 +15,13 @@ public function index(Request $request)
             ->orderBy('general_expenses.created_at', 'desc')
             ->select('general_expenses.*');
-
+            
```
Aqui dejaste mucho espacio

SJAmanda replied:
Resuelto

And-Y21 commented on app/Http/Controllers/MovementHistoryController.php:
```diff
@@ -16,62 +16,146 @@ class MovementHistoryController extends Controller
+            } else {
+                return null;
+            }
+        }
```
Hay que procurar no usar elsif ni if - else

SJAmanda replied:
Resuelto

And-Y21 commented on resources/views/localities/index.blade.php:
```diff
@@ -137,6 +135,54 @@
                                             @include('localities.mailConfiguration')
                                             @include('localities.editPdfBackground')
                                             @include('localities.tokenModal')
+                                            <div class="modal fade" id="historyModal{{ $locality->id }}" tabindex="-1" role="dialog" aria-labelledby="historyModalLabel{{ $locality->id }}" aria-hidden="true">
```
Este modal deberia estar en un archivo aparte, revisa por ejemplo el de pagos por toma de agua

SJAmanda committed:
Refactored MovementHistoryController structure and moved the history modal to a separate Blade file

SJAmanda commented on app/Http/Controllers/DebtController.php:
```diff
@@ -31,12 +32,16 @@ public function index(Request $request)
+        $debts = Debt::with(['waterConnection.customer', 'creator'])
+                ->where('debts.locality_id', $localityId)
+                ->whereHas('creator', function ($query) use ($localityId) {
+                        $query->where('locality_id', $localityId);
```
Claro, no vuelve a pasar

And-Y21 commented on resources/views/reports/pdfMovementsHistoryModuleTitle.blade.php:
No puedes hacer unos solo para ambos casos?

SJAmanda replied:
Resuelto

And-Y21 commented on app/Http/Controllers/MovementHistoryController.php:
```diff
@@ -3,11 +3,7 @@
-use App\Models\Cost;
-use App\Models\GeneralExpense;
-use App\Models\Locality;
+use App\Models\{Payment, Debt, Cost, GeneralExpense, Locality};
```
?

SJAmanda replied:
Resuelto

SJAmanda committed:
Complete refactor of the PDF movement history report system.
Merged three separate Blade templates into a single unified view and optimized the controller for improved maintainability and consistency.

And-Y21 commented on resources/views/reports/pdfMovementsHistory.blade.php:
```diff
@@ -130,45 +156,109 @@
                 </p>
             </div>
-        </div>
+        </div>        
```
Hay que quitar los espacio de aqui

SJAmanda committed:
Fixed controller logic and removed extra spacing in index and pdfMovementsHistory views.

SJAmanda commented on app/Http/Controllers/PaymentController.php:
```diff
@@ -15,8 +15,12 @@
-            ->where('locality_id', $authUser->locality_id);
+        $query = Payment::with(['debt.customer', 'creator'])
+        ->where('locality_id', $authUser->locality_id)
+        ->whereHas('creator', function ($q) use ($authUser) {
```
Resuelto

SJAmanda commented on app/Http/Controllers/MovementHistoryController.php:
```diff
@@ -110,11 +184,11 @@ public function generatePDF(Request $request)
+            'groupedByDay' => $formattedGroupedByDay,
+            'modulesRequested' => $modulesRequested,
+            'moduleNames' => $moduleNames,
+            'weekDays' => $weekDays,
```
Resuelto
